include/fiction/algorithms/simulation/sidb/critical_temperature.hpp

```diff
@@ -44,6 +44,8 @@ namespace fiction
 
 /**
  * This struct stores the parameters for the `critical_temperature` algorithm.
+ *
+ * @tparam TT The type of the truth table specifying the gate behavior.
  */
 template <typename TT>
 struct critical_temperature_params
```

Comment:
Something doesn't feel right here. You have the toggle in the params to do gate-based or non-gate-based simulation, yet even in the non-gate-based case, you must specify a truth table type... Why not have two different calling functions just like we're doing for the operational domain for instance? One receives a truth table, the other doesn't.

Reply:
@marcelwa Absolutely. I will fix it. Thanks for pointing this out.